process/block/shardblock.go

```diff
@@ -1772,16 +1806,27 @@ func (sp *shardProcessor) computeExistingAndRequestMissingMetaHeaders(header dat
 		if hdr.Nonce > sp.hdrsForCurrBlock.highestHdrNonce[core.MetachainShardId] {
 			sp.hdrsForCurrBlock.highestHdrNonce[core.MetachainShardId] = hdr.Nonce
 		}
+
+		if sp.shouldConsiderProofsForNotarization(hdr) {
+			notarizedMetaHdrsBasedOnProofs++
+
+			hasProofForShardHdr := sp.proofsPool.HasProof(core.MetachainShardId, metaBlockHashes[i])
+			sp.hdrsForCurrBlock.hdrHashAndInfo[string(metaBlockHashes[i])].hasProof = hasProofForShardHdr
+			if !hasProofForShardHdr {
+				sp.hdrsForCurrBlock.missingHdrsProofs++
```

Comment:
maybe add the header hash in a map for missing proofs instead?

Reply:
moved the logic to be based on hasProof field, which is part of a map too